spark-plugins/src/test/java/co/cask/hydrator/plugin/spark/test/DecisionTreeRegressionTest.java

```diff
@@ -0,0 +1,338 @@
+    expected.add(new Flight(4, 6, 1.0, "N327AA", 1, 12478, "JFK", 12892, "LAX", 900, 1005.0, 65.0, 1225.0, 1324.0, 59.0,
+                            385.0, 2475, 1.0));
+    expected.add(new Flight(29, 3, 1.0, "N355AA", 2407, 12892, "LAX", 11298, "DFW", 1025, 1023.0, 0.0, 1530.0, 1523.0,
+                            0.0, 185.0, 1235, 0.0));
+    expected.add(new Flight(4, 6, 3.0, "N14991", 6159, 13930, "ORD", 13198, "MCI", 2030, 2118.0, 48.0, 2205.0, 2321.0,
+                            76.0, 95.0, 403, 1.0));
+    expected.add(new Flight(25, 6, 2.0, "N0EGMQ", 3419, 10397, "ATL", 12953, "LGA", 1150, 1229.0, 39.0, 1359.0, 1448.0,
+                            49.0, 129.0, 762, 0.0));
+    expected.add(new Flight(2, 4, 4.0, "N919DE", 1908, 13930, "ORD", 11433, "DTW", 1641, 1902.0, 141.0, 1905.0, 2117.0,
+                            132.0, 84.0, 235, 1.0));
+    expected.add(new Flight(2, 4, 4.0, "N933DN", 1791, 10397, "ATL", 15376, "TUS", 1855, 2014.0, 79.0, 2108.0, 2159.0,
+                            51.0, 253.0, 1541, 1.0));
+    Assert.assertEquals(expected, results);
+  }
+
+  @Test
+  public void testInvalidPredictionField() throws Exception {
+    Map<String, String> properties = new ImmutableMap.Builder<String, String>()
+      .put("fileSetName", "decision-tree-regression-model")
+      .put("path", "decisionTreeRegression")
+      .put("featuresToInclude", "dofM,dofW,scheduleDepTime,scheduledArrTime,carrier,elapsedTime,originId,destId")
+      .put("labelField", "dealyed")
+      .put("maxBins", "100")
+      .put("maxDepth", "9")
+      .build();
+
+    ETLBatchConfig etlConfig = ETLBatchConfig.builder("* * * * *")
+      .addStage(new ETLStage("source", MockSource.getPlugin("flightRecords")))
+      .addStage(new ETLStage("customsink", new ETLPlugin(DecisionTreeTrainer.PLUGIN_NAME, SparkSink.PLUGIN_TYPE,
+                                                         properties, null)))
+      .addConnection("source", "customsink")
+      .build();
+
+    AppRequest<ETLBatchConfig> appRequest = new AppRequest<>(DATAPIPELINE_ARTIFACT, etlConfig);
+    ApplicationId appId = NamespaceId.DEFAULT.app("DecisionTreeTest");
+    ApplicationManager appManager = deployApplication(appId.toId(), appRequest);
+
+    // send records from sample data to train the model
+    List<StructuredRecord> messagesToWrite = new ArrayList<>();
+    messagesToWrite.addAll(getInputData());
+
+    // write records to source
+    DataSetManager<Table> inputManager = getDataset(Id.Namespace.DEFAULT, "flightRecords");
+    MockSource.writeInput(inputManager, messagesToWrite);
+
+    // manually trigger the pipeline
+    WorkflowManager workflowManager = appManager.getWorkflowManager(SmartWorkflow.NAME);
+    workflowManager.start();
+    workflowManager.waitForFinish(5, TimeUnit.MINUTES);
+
+    Assert.assertEquals("FAILED", workflowManager.getHistory().get(0).getStatus().name());
+  }
+
+  @Test
+  public void testInvalidFeatures() throws Exception {
```

Comment:
is this just testing the validate methods in the config? If so, it's better to just 'new' a Config object and call the validate method(s).  That is much faster than this, which requires creating a cdap application and running a workflow.